storage/client_test.go

```diff
@@ -54,16 +54,18 @@ func createTestStore(t *testing.T) (*storage.Store, *util.Stopper) {
 	return createTestStoreWithEngine(t,
 		engine.NewInMem(proto.Attributes{}, 10<<20),
 		hlc.NewClock(hlc.NewManualClock(0).UnixNano),
-		true)
+		true, nil)
 }
 
 // createTestStoreWithEngine creates a test store using the given engine and clock.
 // The caller is responsible for closing the store on exit.
 func createTestStoreWithEngine(t *testing.T, eng engine.Engine, clock *hlc.Clock,
-	bootstrap bool) (*storage.Store, *util.Stopper) {
+	bootstrap bool, ctx *storage.StoreContext) (*storage.Store, *util.Stopper) {
 	stopper := util.NewStopper()
 	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), security.LoadInsecureTLSConfig(), stopper)
-	ctx := storage.TestStoreContext
+	if ctx == nil {
+		ctx = &storage.TestStoreContext
```

Comment:
We need to make a copy here (otherwise the following `ctx.Gossip` line is modifying the global TestStoreContext).
```
ctx = &storage.StoreContext{}
*ctx = storage.TestStoreContext
```